### PR TITLE
Fix queue commands for Plane/Heli 

### LINKED ISSUE #6161: Qeueu multiple commands for helo/planes not working
Queuing of multiple command using the Shift key is not working for planes and helo's. They immediately execute the order that should be queued. 

Infantry and land units work correctly. Using latest release, noticed in RA.

<bountysource-plugin>
---
Want to back this issue? **[Post a bounty on it!](https://www.bountysource.com/issues/3465846-qeueu-multiple-commands-for-helo-planes-not-working?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github)** We accept bounties via [Bountysource](https://www.bountysource.com/?utm_campaign=plugin&utm_content=tracker%2F36085&utm_medium=issues&utm_source=github).
</bountysource-plugin>

### PR DESCRIPTION
**Pull request by hktk1f40**
Feels a little bit like a poor man's solution because of the subclassing but it works. Fixes #6161.

### CODE REVIEW

File: OpenRA.Mods.Common/Activities/Air/Fly.cs [hunk 1]
```diff
@@ -84,4 +84,30 @@ public override IEnumerable<Target> GetTargets(Actor self)
 			yield return target;
 		}
 	}
+
+	public class FlyAndContinueWithCirclesWhenIdle : Fly
+	{
+		public FlyAndContinueWithCirclesWhenIdle(Actor self, Target t)
+			: base(self, t)
+		{
+		}
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Activities/Air/Fly.cs]: `: base (self, t) { }`, please.

File: OpenRA.Mods.Common/Activities/Air/Fly.cs [hunk 2]
```diff
@@ -84,4 +84,30 @@ public override IEnumerable<Target> GetTargets(Actor self)
...
+
+		public FlyAndContinueWithCirclesWhenIdle(Actor self, Target t, WDist minRange, WDist maxRange)
+			: base(self, t, minRange, maxRange)
+		{
+		}
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Activities/Air/Fly.cs]: Same here.

File: OpenRA.Mods.Common/Activities/Air/HeliFly.cs [hunk 3]
```diff
@@ -91,5 +91,33 @@ public override IEnumerable<Target> GetTargets(Actor self)
 		{
 			yield return target;
 		}
+    }
+
+	public class HeliFlyAndLandWhenIdle : HeliFly
+	{
+		private readonly HelicopterInfo info;
+
+        public HeliFlyAndLandWhenIdle(Actor self, Target t, HelicopterInfo info)
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Activities/Air/HeliFly.cs]: Spaces -> Tabs.

File: OpenRA.Mods.Common/Activities/Air/HeliFly.cs [hunk 4]
```diff
@@ -91,5 +91,33 @@ public override IEnumerable<Target> GetTargets(Actor self)
 		{
 			yield return target;
 		}
+    }
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Activities/Air/HeliFly.cs]: Same here.

**Comment by abcdefg30** [comment]
Seems to work good.

Except when you queue a reload activity, all other queued activities are canceled and the reload is done.
Not sure if you wanted to fix that here.

File: OpenRA.Mods.Common/Activities/Air/HeliFly.cs [hunk 5]
```diff
@@ -91,5 +91,33 @@ public override IEnumerable<Target> GetTargets(Actor self)
...
+		{
+			var activity = base.Tick(self);
+
+			if (activity == null && !IsCanceled && info.LandWhenIdle)
+			{
+				var sequence = new List<Activity>();
+				if (info.TurnToLand)
+					sequence.Add(new Turn(self, info.InitialFacing));
+				sequence.Add(new HeliLand(self, true));
+				self.QueueActivity(Util.SequenceActivities(sequence.ToArray()));
```
Comment (abcdefg30) [inline OpenRA.Mods.Common/Activities/Air/HeliFly.cs]: Wouldn't
```
if (info.TurnToLand)
	self.QueueActivity(new Turn(self, info.InitialFacing));

self.QueueActivity(new HeliLand(self, true));
```
work, too?

**Comment by atimoschenkow** [comment]
Enter/Reloading should be ok now too, had to modify the TakeOff activity too for it to work.

**Comment by penev92** [comment]
Seems to work as advertised. Thanks :+1:

**Comment by abcdefg30** [comment]
:+1: / :white_check_mark:

**Comment by abcdefg30** [comment]
[Changelog](https://github.com/OpenRA/OpenRA/wiki/Changelog-(bleed)/_compare/344a8ed9dd3f8559c2387e0a38be006771fc6663...d1bfc0a0c494c3913003474636ef00a6baf14899)

**Pull request merged**